snapshot-labs/snapshot-sequencer
# feat: Save plugin quorum to proposal

Fixes https://github.com/snapshot-labs/pitches/issues/59

Calculates and saves quorum required on proposal itself, so UI doesn't have to calculate everytime someone open a proposal

### How to test:
Update your space settings to have quorum plugin with following different inputs and try creating proposals, then the proposal should contain `quorum` with calculate value

for `static` strategy
```json
{
"total":1234,"strategy":"static"
}
```

for `balance` strategy
```json
{
    "address": "0x3e8c72655e48591d93e6dfda16823db0ff23d859",
    "decimals": 18,
    "strategy": "balance",
    "methodABI": {
      "name": "totalSupply",
      "type": "function",
      "inputs": [],
      "outputs": [
        {
          "name": "",
          "type": "uint256",
          "internalType": "uint256"
        }
      ],
      "stateMutability": "view"
    },
    "quorumModifier": 0.1
  }
```

for multibalance
```JSON
{
  "network": "1",
  "strategy": "multichainBalance",
  "strategies": [
    {
      "symbol": "sveSTG",
      "address": "0x9485DbDa44B279311e3eEe374CED60b5364A97d9",
      "network": "1",
      "decimals": 18,
      "methodABI": {
        "name": "totalSupply",
        "type": "function",
        "inputs": [],
        "outputs": [
          {
            "name": "",
            "type": "uint256",
            "internalType": "uint256"
          }
        ],
        "stateMutability": "view"
      }
    },
    {
      "symbol": "veSTG",
      "address": "0x0e42acBD23FAee03249DAFF896b78d7e79fBD58E",
      "network": "1",
      "decimals": 18,
      "methodABI": {
        "name": "totalSupply",
        "type": "function",
        "inputs": [],
        "outputs": [
          {
            "name": "",
            "type": "uint256",
            "internalType": "uint256"
          }
        ],
        "stateMutability": "view"
      }
    }
  ],
  "quorumModifier": 0.025
}
```
TODO
- [ ] Add tests for `getQuorum`
